components/elm/src/main/ncdio_pio.F90.in

```diff
@@ -1509,9 +1510,15 @@ end subroutine ncd_io_{DIMS}d_{TYPE}_glob
              start(:) = 1 ; count(:) = 1
              call scam_field_offsets(ncid,elmlevel,vardesc,start,count)
              if (present(nt)) then
-               n=3
-               if (start(3) .eq. 0) n=2
-               start(n) = nt ; count(n) = 1
+                !find first non-zero occurrence in the "start" array
+                n = 3
+                do while(start(n) == 0)
+                   n = n - 1
+                enddo
+
+                !time index should replace the first occurrence of zero (after a non-zero occurrence) in the "start" array
+                n_time_ind = n + 1
+                start(n_time_ind) = nt ; count(n_time_ind) = 1
```

Comment:
The logic in the new code is different when start(3) != 0 (the old code uses n_time_ind = 3 in this case while the new code would use n_time_ind = 4)

Reply:
Thanks @jayeshkrishna . I have fixed it now based on your recommendation.